### PR TITLE
Middleman Postprocessor and other Autotagger improvements

### PR DESCRIPTION
**Pull request by ThereforeGames**
Hi,

This PR introduces a number of improvements for the Tagger's RPC service, mostly notably with the addition of a "Middleman postprocessor." To use the it, create `middleman.json` in your `interrogator_rpc` directory.

The feature provides a substantial degree of user control over returned tags, per request #87.

In `middleman.json`, the JSON key is a tag name, and the value is a dictionary of settings. For example, `"handshake":{"name":"shaking_hands","post":0.75}` will replace the `handshake` tag with `shaking_hands` and force its confidence value to 0.75 (which in turn determines its placement on the tag list.)

Here is a complete list of the available settings (**EDIT** - Docs updated for v0.2.0):

- `name`: The new name of the tag that will be returned to the client
- `pre`: The new confidence value of the tag, applied before the threshold check
- `post`: The new confidence value of the tag, applied after the threshold check (if it passes)
- `threshold`: The threshold value to compare the tag's confidence value against, useful if the sensitivity of a certain tag doesn't play nicely with your global threshold setting
- `ban`: If `true`, the tag will be removed from the list of tags that will be returned to the client
- `aliases`: A list of additional tags to return to the client if this tag passes
- `excludes`: A list of tags to prohibit if this tag passes

Additionally, a few special keys are supported:

- `_BLACKLIST_TAGS` takes a list of tag names to prohibit, e.g. `"_BANNED_TAGS":["handshake","dancing"]`
- `_WHITELIST_TAGS` takes a list of allowed tag names; all other tag names will be prohibited
- `_DEBUG_TAGS` takes a list of tags to print information about in the console
- `_UNPROMPTED` takes a path to a copy of [Unprompted](https://github.com/ThereforeGames/unprompted), my templating language that transforms plaintext with complex operations, e.g. `"handshake":{"confidence":"[random _min=0.5 _max=0.9]"}` would allow you to set the confidence of `handshake` to a random number between 0.5 and 0.9
- `_INIT` is arbitrary text to process with Unprompted at the start of the run, you can use it to process your own custom tags

If using Unprompted, you can reference the original tag properties with `[get]`, e.g. `"handshake":{"confidence":"[max '{get confidence}' 0.75]"}`, which will cap the returned confidence to 0.75.

Basically, you can exert your dominance over the Autotagger now, any way you see fit.

---

Full PR changelog:

### Added
- Support for `middleman.json` that enables postprocessing of the returned tags
- `CHANGELOG.md` to track changes
- `run.bat` for Windows users
- Global `DEBUG` flag in for diagnostics in lieu of a proper logger

### CODE REVIEW

**Comment by starik222** [comment]
Thanks for your work! Fixing the discovered bugs took a lot of time, so I was only now able to see your improvements.
